pkg/api/server/v1alpha2/auth/rbac.go

```diff
@@ -120,7 +122,8 @@ func (r *RBAC) Check(ctx context.Context, namespace, resource, verb string) erro
 		if impersonator != nil {
 			if err := impersonator.Check(ctx, r.authz, user); err != nil {
 				log.Println(err)
-				return status.Error(codes.Unauthenticated, "permission denied")
+				retMsg := fmt.Sprintf("%s: %s", retMsg, err.Error())
```

Comment:
nit: We don't need to initialize a new variable. We can reuse, retMsg.
```suggestion
				retMsg = fmt.Sprintf("%s: %s", retMsg, err.Error())
```

Reply:
updated